tst/passthrough-fuse/winposix.c

```diff
@@ -163,25 +186,48 @@ char *realpath(const char *path, char *resolved)
         errno = maperror(err);
         result = 0;
     }
-
     return result;
 }
 
 int uncpath(const char *path, WCHAR *buf, int nchar)
 {
-    if (4 < nchar &&
-        0 < MultiByteToWideChar(CP_UTF8, 0, path, -1, &buf[4], nchar - 4))
+    PWSTR BufP = 0;
+    if ('\\' == path[0] && '\\' == path[1])
+    {
+        if (8 < nchar &&
+            0 < MultiByteToWideChar(CP_UTF8, 0, &path[2], -1, &buf[8], nchar - 8))
+        {
+            BufP = &buf[8];
+            buf[0] = L'\\';
+            buf[1] = L'\\';
+            buf[2] = L'?';
+            buf[3] = L'\\';
+            buf[4] = L'U';
+            buf[5] = L'N';
+            buf[6] = L'C';
+            buf[7] = L'\\';
+        }
+    }
+    else
+    {
+        if (4 < nchar &&
+            0 < MultiByteToWideChar(CP_UTF8, 0, path, -1, &buf[4], nchar - 4))
+        {
+            BufP = &buf[4];
+            buf[0] = L'\\';
+            buf[1] = L'\\';
+            buf[2] = L'?';
+            buf[3] = L'\\';
+        }
+    }
+
+    if (BufP)
     {
-        buf[0] = L'\\';
-        buf[1] = L'\\';
-        buf[2] = L'?';
-        buf[3] = L'\\';
-        int i = 4;
-        while (buf[i])
+        while (*BufP)
```

Comment:
We might want to consider fixing the double slash/backslash problem in this loop. From the `realpath` review comment:

> A problem with `realpath` is that it does not strip trailing backslashes when given a directory like `C:\Path\To\Dir\` or `\\Server\Share\Path\To\Dir\`. This was the case prior to your patches, but it has become a problem now that we use the `\\?\` prefix to access files.
>
> The problem is that when FUSE sends a path like `/`, we append it to `C:\Path\To\Dir\` and get `C:\Path\To\Dir\/`, which later becomes `\\?\C:\Path\To\Dir\\`. Unfortunately the [rules](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#win32-file-namespaces) for the `\\?\` prefix effectively make this path illegal (because the double backslash is passed as it to NTFS which rejects it).

Reply:
Updated PR.